src/services/cherry-picker.ts

```diff
@@ -253,23 +272,26 @@ export class CherryPicker {
         medianSuccessLatency: elapsedTime.toFixed(5),
         weightedSuccessLatency: elapsedTime.toFixed(5),
       }
-
-      // Set expiry on relayTiming log
-      await this.redis.expire(blockchainID + '-' + id + '-relayTiming', 300)
     }
 
     await this.redis.set(blockchainID + '-' + id + '-service', JSON.stringify(serviceQuality), 'EX', ttl)
   }
 
-  sortAndBucketArray(raw: string[]): SortedServiceQuality {
-    // Convert redis items into floats
-    const convertedRaw = raw.map(function (x) {
-      return parseFloat(x)
-    })
-    const sorted = convertedRaw.sort((a, b) => a - b)
+  reduceArray(raw: number[]): number[] {
+    // Remove every 2nd member of array
+    if (raw.length > 500) {
+      const reducedArray = raw.filter(function (_, i) {
+        return (i + 1) % 2
```

Comment:
I think this is doing the opposite of what's intended? This is removing every _odd_ element of the array, instead of every even element.

Reply:
Also, what's the logic behind the 500? About the number, not why.

Reply:
Haha I don't really care if it's odd or even, I just want a reduction in array size while retaining the distribution.

500 seems like a large enough array of elapsed times to generate a good picture of a node's response while not making excessively long redis keys or arrays to sort.

Reply:
Ohh gotcha haha. Then yeah, this will work well, the comment just stood out to me. If we wanted to be "mathy/smart" about it we could also reduce the array by averaging adjacent values together, but I'm not necessarily sure this would perform better than just doing this :D.

Sounds good!